payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -82,82 +83,55 @@ internal class GooglePayLauncherActivity : AppCompatActivity() {
         fadeOut()
     }
 
-    private fun payWithGoogle(task: Task<PaymentData>) {
-        AutoResolveHelper.resolveTask(
-            task,
-            this,
-            LOAD_PAYMENT_DATA_REQUEST_CODE
-        )
-    }
+    private fun onGooglePayResult(taskResult: ApiTaskResult<PaymentData>) {
+        when (taskResult.status.statusCode) {
+            CommonStatusCodes.SUCCESS -> {
+                val paymentDataJson = JSONObject(taskResult.result!!.toJson())
```

Comment:
what's the impact of result being null? Will we crash?

Reply:
(could be fixed by sharing more code with GooglePayPaymentMethodLauncherActivity, see my comment there)